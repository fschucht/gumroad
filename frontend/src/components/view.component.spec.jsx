import renderer from 'react-test-renderer';
import { View } from './view.component';

describe('View', () => {
  it('should render a View component', () => {
    const renderedComponent = renderer.create(<View>Hello World!</View>).toJSON()

    expect(renderedComponent).toMatchSnapshot()
  })

  describe('when hasBorderTop is true', () => {
    it('should render a View component with a top border', () => {
      const renderedComponent = renderer.create(<View hasBorderTop>Hello World!</View>).toJSON()

      expect(renderedComponent).toMatchSnapshot()
    })
  })

  describe('when hasBorderLeft is true', () => {
    it('should render a View component with a left border', () => {
      const renderedComponent = renderer.create(<View hasBorderLeft>Hello World!</View>).toJSON()

      expect(renderedComponent).toMatchSnapshot()
    })
  })

  describe('when hasBorderRight is true', () => {
    it('should render a View component with a right border', () => {
      const renderedComponent = renderer.create(<View hasBorderRight>Hello World!</View>).toJSON()

      expect(renderedComponent).toMatchSnapshot()
    })
  })

  describe('when hasBorderBottom is true', () => {
    it('should render a View component with a bottom border', () => {
      const renderedComponent = renderer.create(<View hasBorderBottom>Hello World!</View>).toJSON()

      expect(renderedComponent).toMatchSnapshot()
    })
  })
})