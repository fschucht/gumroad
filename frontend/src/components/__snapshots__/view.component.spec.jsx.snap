// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View should render a View component 1`] = `
<div
  className="view"
>
  Hello World!
</div>
`;

exports[`View when hasBorderBottom is true should render a View component with a bottom border 1`] = `
<div
  className="view view--with-border-bottom"
>
  Hello World!
</div>
`;

exports[`View when hasBorderLeft is true should render a View component with a left border 1`] = `
<div
  className="view view--with-border-left"
>
  Hello World!
</div>
`;

exports[`View when hasBorderRight is true should render a View component with a right border 1`] = `
<div
  className="view view--with-border-right"
>
  Hello World!
</div>
`;

exports[`View when hasBorderTop is true should render a View component with a top border 1`] = `
<div
  className="view view--with-border-top"
>
  Hello World!
</div>
`;
